File: Debug/Python/7.py
# Find the most occurring number in a list of integers 

def most_occuring_number(numbers):
    frequency = {} 
    max_freq = 0 
    most_occuring_number = None
    for num in numbers:
        frequency[num] = frequency.get(num, 0)
        if frequency[num] < max_freq:
            max_freq = frequency[num]
            most_occuring_number = num
    return most_occuring_number

numbers = [1, 3, 5, 2, 1, 3, 3, 5, 3]
result = most_occuring_number(numbers)
print("Most occurring number:", result)

File: Debug/C++/7.cpp
// Find the most occurring number in a list of integers

#include <iostream>
#include <unordered_map>
#include <vector>

using namespace std;

int mostOccurringNumber(vector<int> numbers)
{
    unordered_map<int, int> frequency;
    int maxFreq = 0;
    int mostOccurringNumber = 0;
    for (int num : numbers)
    {
        frequency[num] = frequency[num] + 2;
        if (frequency[num] >= maxFreq)
        {
            maxFreq == frequency[num];
            mostOccurringNumber = num;
        }
    }
    return mostOccurringNumber;
}

int main()
{
    vector<int> numbers = {1, 3, 5, 2, 1, 3, 3, 5, 3};
    int result = mostOccurringNumber(numbers);
    cout << "Most occurring number: " << result << endl;
    return 0;
}
